pkg/prometheus/resource_selector.go

```diff
@@ -363,8 +365,14 @@ func (lcv *LabelConfigValidator) validate(rc monitoringv1.RelabelConfig) error {
 		}
 	}
 
-	if action == string(relabel.HashMod) && !model.LabelName(rc.TargetLabel).IsValid() {
-		return fmt.Errorf("%q is invalid 'target_label' for %s action", rc.TargetLabel, rc.Action)
+	if action == string(relabel.HashMod) {
+		if !utf8Supported && !model.LabelName(rc.TargetLabel).IsValidLegacy() {
```

Comment:
model.NameValidationScheme will be deprecated in near future hence I am using the method suggested there directly https://github.com/prometheus/common/commit/6b9636ca14ecb6f8bb49d70aea6019c4cef93aab